### PR TITLE
Dragonwrath Pack commit.

### PR DESCRIPTION
**Pull request by LyrenScalari**

### CODE REVIEW

File: src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+    // Actions the card should do.
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        CardCrawlGame.sound.play("POWER_MANTRA", 0.05F);
+        AbstractDungeon.actionManager.addToBottom(new SFXAction("ORB_LIGHTNING_EVOKE"));
+        addToBot(new VFXAction(new LightningEffect(p.drawX,p.drawY)));
+        addToBot(new DamageCallbackAction(p, new DamageInfo(p,magicNumber, DamageInfo.DamageType.THORNS), AbstractGameAction.AttackEffect.NONE, integer -> {
+            System.out.println(integer);
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java:53]: no debug logging please

File: src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+
+    // Actions the card should do.
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        CardCrawlGame.sound.play("POWER_MANTRA", 0.05F);
+        AbstractDungeon.actionManager.addToBottom(new SFXAction("ORB_LIGHTNING_EVOKE"));
+        addToBot(new VFXAction(new LightningEffect(p.drawX,p.drawY)));
+        addToBot(new DamageCallbackAction(p, new DamageInfo(p,magicNumber, DamageInfo.DamageType.THORNS), AbstractGameAction.AttackEffect.NONE, integer -> {
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java:52]: Why do you do a damage callback action here?

File: src/main/java/thePackmaster/cards/dragonwrathpack/SearingLight.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+
+    // Actions the card should do.
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        CardCrawlGame.sound.play("POWER_MANTRA", 0.05F);
+        AbstractDungeon.actionManager.addToBottom(new SFXAction("ORB_LIGHTNING_EVOKE"));
+        addToBot(new VFXAction(new LightningEffect(p.drawX,p.drawY)));
+        addToBot(new DamageCallbackAction(p,new DamageInfo(p,magicNumber, DamageInfo.DamageType.THORNS), AbstractGameAction.AttackEffect.NONE, integer -> {
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/dragonwrathpack/SearingLight.java:56]: again confused about the callback

File: src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.helpers.FontHelper;
+import com.megacrit.cardcrawl.localization.PowerStrings;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import thePackmaster.actions.dragonwrathpack.SmiteAction;
+
+import static thePackmaster.SpireAnniversary5Mod.makeID;
+
+public class PenancePower extends TwoAmountPower implements CloneablePowerInterface, HealthBarRenderPower {
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java:18]: packmaster's power abstract already extends twoamount

File: src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        this.loadRegion("heartDef");
+
+        updateDescription();
+    }
+    @Override
+    public void onInitialApplication() {
+        amount2 = Power;
+        updateDescription();
+        if (this.amount >= 8){
+            amount -= 8;
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java:49]: Might want to remove the power if it's at 0 now

File: src/main/java/thePackmaster/powers/dragonwrathpack/confessionpower.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        // We load those textures here.
+        this.loadRegion("mantra");
+
+        updateDescription();
+    }
+
+    public int onAttacked(DamageInfo info, int damageAmount) {
+        CardCrawlGame.sound.play("POWER_MANTRA", 0.05F);
+        for (AbstractMonster m : AbstractDungeon.getCurrRoom().monsters.monsters){
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/dragonwrathpack/confessionpower.java:49]: Please use Wiz.getEnemies(), this doesn't filter out dead enemies

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/actions/dragonwrathpack/SmiteAction.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            return;
+        }
+        if (!shot) {
+            AbstractDungeon.effectList.add(new SmiteEffect(target, this));
+            for (int i = 0; i < AbstractDungeon.miscRng.random(20, 30); ++i) {
+                AbstractDungeon.effectsQueue.add(new DivineEyeParticle());
+            }
+            shot = true;
+        }
+        if (doDamage && !target.isDeadOrEscaped()) {
```
[the remark this answers is not in the document]
Comment (LyrenScalari) [reply src/main/java/thePackmaster/actions/dragonwrathpack/SmiteAction.java:40]: doDamage is flipped as part of the effect it queues. it's specifically to delay the damage so it lines up with the animation

File: src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java (hunk 2 above)
Comment (LyrenScalari) [reply src/main/java/thePackmaster/cards/dragonwrathpack/SacredTrial.java:52]: old code. copied from Dragonkin which used to have reasons to do that. won't be there once the balance is done.

File: src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java [hunk 8]
```diff
@@ -50,6 +51,9 @@ public void onInitialApplication() {
             CardCrawlGame.sound.play("POWER_MANTRA", 0.05F);
             addToBot(new SmiteAction(owner,new DamageInfo(owner,Power, DamageInfo.DamageType.HP_LOSS)));
             Power += 10;
+            if (amount > 1){
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/dragonwrathpack/PenancePower.java:54]:
```suggestion
            if (amount < 1){
```

**Pull request merged**